### ISSUE TITLE
Bug: (sort-imports) side-effect imports with an internal pattern are defined as internal modules

### ISSUE THREAD

**Issue by Wondermarin** [azat-io/eslint-plugin-perfectionist#72]
### Describe the bug

If a side-effect import starts with an internal pattern, then it will be defined as an internal module and not a side-effect.

### Code example

**IN CONFIG:**
```json
"perfectionist/sort-imports": [
  "error",
  {
    "type": "line-length",
    "order": "desc",
    "internal-pattern": ["@/**"],
    "groups": [
      "builtin",
      "external",
      "internal",
      "parent",
      "sibling",
      "index",
      "side-effect"
    ]
  }
],
```

**IN CODE:**
```jsx
import { App } from "@/components/app"; // ⤸ There is no space between groups.
import "@/styles/globals.css";
```

### ESLint version

v8.49.0

### ESLint Plugin Perfectionist version

v2.1.0

### Additional comments

Starting from the idea of the plugin, I think that the current behavior is not quite correct, because there is a group for side-effect imports and therefore something like this is not very aesthetically?
```js
import { EvenLongerNameComponent } from "@/components";
import { LongNameComponent } from "@/components";
import { AnotherComponent } from "@/components";
import { Component } from "@/components";
import "@/styles/globals.css";
import { s } from "@/css";

import "./styles.css";
import "./base.css";
```

### Validations

- [X] Read the [docs](https://eslint-plugin-perfectionist.azat.io).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by Wondermarin** [comment]
Made [PR](https://github.com/azat-io/eslint-plugin-perfectionist/pull/73).

**Comment by Wondermarin** [comment]
The problem is not critical, you can create a release as it will be more convenient for you.

**Comment by azat-io** [comment]
Released in [v2.2.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v2.2.0).
